common/folderDeletionManager.go

```diff
@@ -155,7 +171,7 @@ func (s *standardFolderDeletionManager) RecordChildDeleted(childFile *url.URL) {
 	}
 
 	s.mu.Lock()
-	folderStatePtr, alreadyKnown := s.contents[folder]
+	folderStatePtr, alreadyKnown := s.contents[s.getMapKey(folder)]
```

Comment:
I think better to break this into two statements and log if getMapKey returns unexpected strings like an empty string or something.

Reply:
I don't see much value in this. "" should be the root directory. A path starting with no slash is weird, but since we're working with parsed URLs here and not constructing them ourselves, I doubt this would occur.